specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimanagement.json

```diff
@@ -0,0 +1,857 @@
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Version of the API to be used with the client request."
+    },
+    "ApiVersionSetIdParameter": {
+      "name": "versionSetId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Api Version Set identifier. Must be unique in the current API Management service instance.",
+      "minLength": 1,
+      "maxLength": 80,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "AppTypeParameter": {
+      "name": "appType",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "description": "Determines the type of application which send the create user request. Default is legacy publisher portal.",
+      "enum": [
+        "portal",
+        "developerPortal"
+      ],
+      "x-ms-enum": {
+        "name": "AppType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "portal",
+            "description": "User create request was sent by legacy developer portal."
+          },
+          {
+            "value": "developerPortal",
+            "description": "User create request was sent by new developer portal."
+          }
+        ]
+      },
+      "default": "portal",
+      "x-ms-parameter-location": "method"
+    },
+    "AttachmentIdParameter": {
+      "name": "attachmentId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Attachment identifier within an Issue. Must be unique in the current Issue.",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "AuthorizationProviderIdParameter": {
+      "name": "authorizationProviderId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Identifier of the authorization provider.",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "AuthorizationIdParameter": {
+      "name": "authorizationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Identifier of the authorization.",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "AuthorizationAccessPolicyIdParameter": {
+      "name": "authorizationAccessPolicyId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Identifier of the authorization access policy.",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
```

Comment:
What are the format or nature of these id parameters - GUID? Resource ID? Something else?